File: arthr.Data/arTask/Timesheet.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace arthr.Data.arTask
{
    
    public class Timesheet
    {

        public int Id { get; set; }
        public int Hours { get; set; }
        public int Mins { get; set; }

        [Display(Name = "Hourly Rate")]
        public int HourlyRate { get; set; }

        [Display(Name = "Date Recorded")]
        public DateTime? DateRecorded { get; set; }

        [DisplayFormat(DataFormatString="{0:C}")]
        public double Quoted { get; set; }

        public bool AlreadyBilled { get; set; }
        
        public int StaffId { get; set; }
        public virtual Core.Staff Staff { get; set; }

        public int AnthRTaskId { get; set; }
        public virtual AnthRTask AnthRTask { get; set; }
        public bool? HideFromTimesheet { get; set; }

    }

}

File: arthr.Web/Scripts/Models/Timesheet.ts
import { AnthRTask } from './AnthRTask';
import { Staff } from './Staff';

export class Timesheet {
    alreadyBilled: boolean;
    anthRTask: AnthRTask;
    anthRTaskId: number;
    dateRecorded: Date;
    hideFromTimesheet: boolean;
    hourlyRate: number;
    hours: number;
    timesheetId: number;
    mins: number;
    quoted: number;
    staff: Staff;
    staffId: number;
}
/*
    Debug Info:
    Unknown Types: AnthRTask, Staff
*/
